src/main/java/spoon/ContractVerifier.java

```diff
@@ -203,6 +203,7 @@ public void checkShadow() {
 			public void scan(CtElement element) {
 				if (element != null && CtShadowable.class.isAssignableFrom(element.getClass())) {
 					assertFalse(((CtShadowable) element).isShadow());
+					return;
```

Comment:
If you do add this `return` here, you prevent the scanner (here `checkShadow`) to visitit descendent of a `CtElement` if it is a class that implement `CtShadowable`. As we run contract versifier on a root package (an instance of `CtPackage` which does implement `CtShadowable`), this patch would stop the visit immediatly, i.e. it stop verifying the contract. While it does make the test pass, it basically remove its body.

I think the root of the bug (at least in the case of `fieldaccesses/testclasses/Tacos`) is that `typeDeclaration` line 218 is null for a reference pointing to `Foo.SubInner`. 
(And AFAIU it should not be as we have the sources for `spoon.test.fieldaccesses.testclasses.internal.Bar.Inner.SubInner`, and `Foo` extends `Bar.Inner`, so  `Foo.SubInner` should exist and be correctly resolved.)